pinot-controller/src/main/java/org/apache/pinot/controller/recommender/data/writer/CsvWriter.java

```diff
@@ -38,6 +42,24 @@ protected String generateRow(DataGenerator generator) {
     return StringUtils.join(values, ",");
   }
 
+  @Override
+  public void write()
+      throws Exception {
+    super.write();
```

Comment:
IIUC, this go over the entire file twice? And repeats this for each file?

Could we avoid this by having the base FileWriter have a `init(FileWriter writer)` function that can be called in the original write loop each time a file is created, and the CSVWriter can override it to add the headers to avoid the 2nd pass?

Reply:
That's right. Made the changes.